public/static-checkout/js/scripts.js

```diff
@@ -0,0 +1,309 @@
+/* eslint-disable vars-on-top */
+/* eslint-disable prefer-template */
+/* eslint-disable prefer-arrow-callback */
+/* eslint-disable no-var */
+var checkoutButton = document.getElementById('place-order-button');
+var paypalButton = document.getElementById('submit-paypal');
+var couponButton = document.getElementById('coupon-button');
+var userButton = document.querySelector('.user-button');
+var countrySelect = document.getElementById('country');
+var stateSelect = document.getElementById('state');
+var countryStates = {
+  CA: {
+    AB: 'Alberta',
+    BC: 'British Columbia',
+    MB: 'Manitoba',
+    NB: 'New Brunswick',
+    NL: 'Newfoundland and Labrador',
+    NS: 'Nova Scotia',
+    NT: 'Northwest Territories',
+    NU: 'Nunavut',
+    ON: 'Ontario',
+    PE: 'Prince Edward Island',
+    QC: 'Québec',
+    SK: 'Saskatchewan',
+    YT: 'Yukon',
+  },
+  US: {
+    AL: 'Alabama',
+    AK: 'Alaska',
+    AZ: 'Arizona',
+    AR: 'Arkansas',
+    AA: 'Armed Forces Americas',
+    AE: 'Armed Forces Europe',
+    AP: 'Armed Forces Pacific',
+    CA: 'California',
+    CO: 'Colorado',
+    CT: 'Connecticut',
+    DE: 'Delaware',
+    DC: 'District Of Columbia',
+    FL: 'Florida',
+    GA: 'Georgia',
+    HI: 'Hawaii',
+    ID: 'Idaho',
+    IL: 'Illinois',
+    IN: 'Indiana',
+    IA: 'Iowa',
+    KS: 'Kansas',
+    KY: 'Kentucky',
+    LA: 'Louisiana',
+    ME: 'Maine',
+    MD: 'Maryland',
+    MA: 'Massachusetts',
+    MI: 'Michigan',
+    MN: 'Minnesota',
+    MS: 'Mississippi',
+    MO: 'Missouri',
+    MT: 'Montana',
+    NE: 'Nebraska',
+    NV: 'Nevada',
+    NH: 'New Hampshire',
+    NJ: 'New Jersey',
+    NM: 'New Mexico',
+    NY: 'New York',
+    NC: 'North Carolina',
+    ND: 'North Dakota',
+    OH: 'Ohio',
+    OK: 'Oklahoma',
+    OR: 'Oregon',
+    PA: 'Pennsylvania',
+    RI: 'Rhode Island',
+    SC: 'South Carolina',
+    SD: 'South Dakota',
+    TN: 'Tennessee',
+    TX: 'Texas',
+    UT: 'Utah',
+    VT: 'Vermont',
+    VA: 'Virginia',
+    WA: 'Washington',
+    WV: 'West Virginia',
+    WI: 'Wisconsin',
+    WY: 'Wyoming',
+  },
+};
+
+var ecommerceBaseUrl;
+if (window.location.hostname === 'payment.edx.org') {
+  ecommerceBaseUrl = 'https://ecommerce.edx.org';
+} else if (window.location.hostname === 'payment.stage.edx.org') {
+  ecommerceBaseUrl = 'https://ecommerce.stage.edx.org';
+} else if (window.location.hostname === 'localhost') {
+  ecommerceBaseUrl = 'http://localhost:18130';
+}
+
+function getBasketId() {
+  var queryString = window.location.search;
+  var urlParams = new URLSearchParams(queryString);
+  // Emma: sql or other injection?
+  var basketId = urlParams.get('basketId');
+  // Emma: what if the basketId is not in the param?
+  return basketId;
+}
+
+function redirectToMFE(couponCode) {
+  var queryString = window.location.search;
```

Comment:
Any variable like `queryString` here that's only being used in one following statement isn't really buying you much but additional bytes.

```
var urlParams = new URLSearchParams(window.location.search);
```